File: demo-user-web/src/main/java/com/demo/user/controller/UserController.java
package com.demo.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.demo.pojo.TbUser;
import com.demo.user.service.UserService;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    @RequestMapping("/findList")
    public List<TbUser> findList() {
      return userService.findUserList();
    }

    @RequestMapping("/findUser")
    public TbUser findUser(String username){
        return userService.findUser(username);
    }

    @RequestMapping("/findUsersByAge")
    public List<TbUser> findUsersByAge(Integer start, Integer end){
        return userService.findUsersByAge(start,end);
    }
}

File: demo-user-web/src/main/webapp/js/controller/userController.js
app.controller("userController",function ($scope,userService) {

    //查询所有用户
    $scope.findList=function () {
        userService.findList().success(function (response) {
            $scope.list=response;
        })
    }

    //根据用户名查询用户信息
    $scope.findUser=function (username) {
        userService.findUser(username).success(function (response) {
            $scope.entity=response;
        })
    }

    //根据年龄段查询用户
    $scope.findUsersByAge=function (start,end) {
        userService.findUsersByAge(start,end).success(function (response) {
            $scope.users=response;
        })
    }
})
